zanata-adapter-xliff/src/main/java/org/zanata/adapter/xliff/XliffWriter.java

```diff
@@ -216,30 +221,12 @@ public static void writeFile(File file, Resource doc, String locale,
             // end xliff tag
             writer.writeEndDocument();
             writer.flush();
+            writer.close();
```

Comment:
Oh nested try yes this will work. I thought you mean adding finally to the
outer try block.

On Thu, Feb 25, 2016 at 11:12 AM, Sean Flanigan <notifications@github.com>
wrote:

> In
> zanata-adapter-xliff/src/main/java/org/zanata/adapter/xliff/XliffWriter.java
> <https://github.com/zanata/zanata-common/pull/19#discussion_r54035260>:
>
> > @@ -216,30 +221,12 @@ public static void writeFile(File file, Resource doc, String locale,
> >              // end xliff tag
> >              writer.writeEndDocument();
> >              writer.flush();
> > +            writer.close();
>
> @huangp <https://github.com/huangp>
> No, try with resources won't close its resources first, as long as the
> try-finally is inside the try with resources.
>
> try (FileOutputStream fileStream = new FileOutputStream(file)) {
>     serializer.setOutputStream(fileStream);
>     writer = serializer.getXMLStreamWriter();
>     try {
>         // use the writer
>     } finally {
>         closeQuietly(writer);
>     }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-common/pull/19/files#r54035260>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com